whd/dispatcher.py

```diff
@@ -211,6 +212,57 @@ def _set_pr_labels(self, pr_event, resources):
                     f" in '{repository_path}' found. Ignoring, since they are not an org member'."
                 )
 
+    def _start_not_triggered_jobs(self, concourse_api, resources):
+        for resource in resources:
+            if resource.type != "pull-request":
+                continue
+
+            resource_versions = concourse_api.resource_versions(
+                resource.pipeline.name,
+                resource.name
+            )
+            if not resource_versions:
+                continue
+
+            jobs_with_triggering_resource = filter(
+                lambda job: job.has_triggering_resource_in_get_step(resource.name),
+                resource.pipeline.jobs
+            )
+            for job in jobs_with_triggering_resource:
+                builds = concourse_api.job_builds(job.pipeline.name, job.name)
+                for resource_version in self._resource_version_ids_not_build(
+                    resource_versions, builds
+                ):
+                    logger.info(
+                        f"Pin resource version {resource_version.version()} "
+                        f"and trigger build for job {job.name}"
+                    )
+                    self._pin_trigger_unpin(job, resource.name, resource_version.id())
+
+    def _resource_version_ids_not_build(self, resource_versions, builds):
+        for resource_version in resource_versions[:10]:
+            resource_version_in_build = False
+            for build in builds[-10:]:
+                build_plan = build.plan()
+                if build_plan.contains_version_ref(resource_version.version()['ref']):
+                    resource_version_in_build = True
+                    break
+            if not resource_version_in_build:
+                yield resource_version
+
+    def _pin_trigger_unpin(job, resource_name, resource_version_id):
```

Comment:
could you please add some conjunctions to the method's name, to make its semantics easier to grasp?
-> pin_and_trigger_and_unpin

or maybe (less lowlevel): trigger_execution_with_pinned_resource_version

also, since those methods (I am referring to most parts of this PR) implement rather complex, and unintuitive semantics, I would very much vote for adding _brief_ doc-strs explaining said semantics